flocker/node/agents/ebs.py

```diff
@@ -128,15 +106,18 @@ def _run_with_logging(self, *args, **kwargs):
         """
         original = getattr(self, original_name)
         method = getattr(original, method_name)
-        try:
-            return method(*args, **kwargs)
-        except EC2ResponseError as e:
-            BOTO_EC2RESPONSE_ERROR(
-                code=e.code,
-                message=e.message,
-                request_id=e.request_id,
-            ).write()
-            raise
+
+        # Trace IBlockDeviceAPI ``method`` as Eliot Action.
+        with AWS_ACTION(operation=[method_name, args, kwargs]):
+            try:
+                return method(*args, **kwargs)
+            except EC2ResponseError as e:
+                BOTO_EC2RESPONSE_ERROR(
+                    aws_code=e.code,
+                    aws_message=e.message,
+                    aws_request_id=e.request_id,
+                ).write()
```

Comment:
> It'd be interesting to see if the new log messages allow us to diagnose why that call to delete_volume fails.
Volume vol-a346f1b3 is currently attached to i-115ccfe6
> Why is that volume still attached?
> Is it because we haven't yet attempted to detach it?
> No, I can see a "successful" detach operation logged prior to calling delete, so perhaps we need to make the detach method more robust.

Yes, the log message pointed to the cause of the failed test. What happened was:

1) Detach succeeded in initiating detach.
2) We timed out (after 180 seconds) waiting for the the volume to transition from ``in-use`` to ``available``. Detach took longer than 180 seconds.
3) Detach operation returned success.
4) Test moved on to attempting destroy while detach was still in progress.